frontend/device/kobo/powerd.lua

```diff
@@ -321,8 +321,9 @@ function KoboPowerD:turnOffFrontlightHW()
     util.runInSubProcess(function()
         for i = 1,5 do
             self:_setIntensity(math.floor(self.fl_intensity - ((self.fl_intensity / 5) * i)))
-            -- NOTE: We generally don't need to sleep when using sysfs as a backend...
-            if self.device:hasNaturalLight() and not self.device:hasNaturalLightMixer() then
+            --- @note: Newer devices appear to block slightly longer on FL ioctls/sysfs, so only sleep on older devices,
+            ---        otherwise we get a jump and not a ramp ;).
+            if not self.device:hasNaturalLight() then
                 if (i < 5) then
                     util.usleep(35 * 1000)
```

Comment:
Btw, how'd you come up with that value?

Reply:
Trial & error, mostly. The goal was having the whole thing stay under a certain total duration (like, 500ms at worst), having enough steps to make it look decent everywhere (barring the ramp-down glitch on < Mk. 7), but not too much that the duration gets blown on slow devices.

Turns out, that's only a tiny bit faster than the delay of Mk. 7 ioctls ;p. Which is what I should have started with, as the ramp looks neat there ;).